src/main/java/com/bootcamp/demo/models/User.java

```diff
@@ -0,0 +1,72 @@
+package com.bootcamp.demo.models;
+
+import java.util.Objects;
+
+public final class User {
+    private final Long userId;
+    private String email;
+    private String firstName;
+    private String lastName;
+
+    public User(Long userId, String email, String firstName, String lastName) {
+        this.userId = userId;
+        this.email = email;
+        this.firstName = firstName;
+        this.lastName = lastName;
+    }
+
+    public Long getUserId() {
+        return userId;
+    }
+
+    public String getEmail() {
+        return email;
+    }
+
+    public String getFirstName() {
+        return firstName;
+    }
+
+    public String getLastName() {
+        return lastName;
+    }
+
+    public User setUserId(Long id) {
+        return new User(id, email, firstName, lastName);
+    }
+
+    public void setEmail(String email) {
+        this.email = email;
+    }
+
+    public void setFirstName(String firstName) {
+        this.firstName = firstName;
+    }
+
+    public void setLastName(String lastName) {
+        this.lastName = lastName;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) return true;
```

Comment:
Some of the issues I now know this use of `if` without braces could pose:
    1. if/else ambiguity - which I didn't consider to be a problem here as the return statement rules out the execution of code beyond that point.
    2. It could cause errors if later on if we decide to comment or add code on a certain `if` branch and don't add the braces.

Reply:
Its a good practice which will prevents future errors.